pnpm/test/switchingVersions.test.ts

```diff
@@ -46,6 +46,19 @@ test('do not switch to pnpm version that is specified not with a semver version'
   expect(stdout.toString()).toContain('Cannot switch to pnpm@kevva/is-positive')
 })
 
+test('do not switch to pnpm version that is specified starting with v', async () => {
+  prepare()
+  const pnpmHome = path.resolve('pnpm')
+  const env = { PNPM_HOME: pnpmHome }
+  writeJsonFile('package.json', {
+    packageManager: 'pnpm@v9.15.5',
+  })
+
+  const { stdout } = execPnpmSync(['help'], { env })
+
+  expect(stdout.toString()).toContain('Cannot switch to pnpm@v9.15.5')
```

Comment:
why only assert part of the error message? "is not a valid version" is the part that's really helpful that it'd be good to check for

Reply:
This is because the previous test case is like this, I refer to the previous test case. As you said, it might be better to complete the error message assertion, I will update it.